routes/x/module.tsx

```diff
@@ -777,6 +819,110 @@ function InfoView(
   );
 }
 
+function generateURL(
```

Comment:
Nice util! But let's move this to `util/` and add some test cases for each registry entry as it's difficult to test each switch branch in UI

Reply:
yep, will do. was actually thinking of moving this to apiland